sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -419,6 +419,81 @@ case class GpuCot(child: Expression) extends GpuUnaryMathExpression("COT") {
   }
 }
 
+case class GpuHypot(left: Expression, right: Expression) extends CudfBinaryMathExpression("HYPOT") {
+
+  override def binaryOp: BinaryOp = BinaryOp.ADD
+
+  // naive implementation of hypot
+  // r = sqrt(lhs^2 + rhs^2) 
+  // This is prone to overflow with regards either square term
+  // However, we can reduce it to
+  // r = sqrt(x^2 + y^2) = x * sqrt(1 + (y/x)^2)
+  // where x = max(abs(lhs), abs(rhs)), y = min(abs(lhs), abs(rhs))
+  // which will only overflow if both terms are near the maximum representation space
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    // Using this form, because we really need a max and min that aren't NULL-aware
+    // In that case, then result will always be NULL, which is consistent with Spark
+    val (x, y) = withResource(Seq(lhs.getBase.abs, rhs.getBase.abs)) { 
```

Comment:
Just like the other comment. There are multiple places in the code where you try to keep the line count small my doing two operations in a Sequence or something similar and then putting both in a `withResource` call to make it safe.  That is technically not safe, because if the second one fails the first one can leak.  We have a `safeMap` API that can handle the corner cases for you, but I personally think it would be cleaner to just have multiple withResource calls in the right places and to split things up into smaller methods where possible.